sheerwater_benchmarking/data/stage_iv.py

```diff
@@ -0,0 +1,123 @@
+"""Stage IV precipitation data product."""
+import numpy as np
+import xarray as xr
+from dateutil import parser
+
+from sheerwater_benchmarking.utils import apply_mask, cacheable, clip_region, dask_remote, regrid, roll_and_agg
+
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['year', 'grid'],
+           chunking={'lat': 300, 'lon': 300, 'time': 365})
+def stage_iv_gridded(year, grid):
+    """Regridded Stage IV precipitation data by year.
+
+    Reads data from USGS OSN zarr store.
+    Data is available from 2002-present.
+    """
+    # Open the zarr store
+    store = 'https://usgs.osn.mghpcc.org/mdmf/gdp/stageiv_combined.zarr/'
+    ds = xr.open_dataset(store, engine='zarr', chunks={})
+
+    # Select the year
+    ds = ds.sel(time=slice(f"{year}-01-01", f"{year}-12-31"))
+
+    # Ensure consistent naming
+    ds = ds.rename(
+        {'Total_precipitation_surface_1_Hour_Accumulation': 'precip'},
+    )
+
+    # chirps is in "units": "mm/day"
```

Comment:
Can delete this comment abou thte CHIRPS dataset, or simply say our target is mm/day units, stage iv is in ....

Reply:
updated